### ISSUE TITLE
`config.json` returns incorrect port while behind a proxy

### ISSUE THREAD

**Issue by chrsmutti** [kellnr/kellnr#53]
I'm trying to deploy a `kellnr` instance behind a proxy that forces `https` traffic.

My current configuration is:

```
KELLNR_API_ADDRESS=example.com
KELLNR_API_PROTOCOL=https
KELLNR_API_PORT=8000
KELLNR_API_PORT_PROXY=443
```

My assumption here is that the `kellnr` instance behind the proxy would communicate with the proxy on port 8000 and users would go through the proxy and requests would be done to `https://example.com:443/api/v1/crates`. The issue is that [when building the `ConfigJson` object only the `api_port` setting is used](https://github.com/kellnr/kellnr/blob/main/crates/index/src/config_json.rs#L37-L47C2).

Am I correct in assuming that it should check if `api_port != api_port_proxy` and use the latter instead?

This is the current response of `/api/v1/crates/config.json` with the configuration above:

```
{
  "dl": "https://example.com:8000/api/v1/crates",
  "api": "https://example.com:8000",
  "auth-required": false
}
```

I'd expect it to be:

```
{
  "dl": "https://example.com:443/api/v1/crates",
  "api": "https://example.com:443",
  "auth-required": false
}
```

**Comment by secana** [comment]
Hi,

I'll look into that. I'm running Kellnr behind a https proxy, but you are right that the `config.json` does not reflect the correct value. I'll fix that asap.

**Comment by secana** [comment]
I released Kellnr 4.0.3-pre1

Can you check if the issue is fixed for you?

**Comment by chrsmutti** [comment]
It did work! Thanks for the quick fix! 😄

**Comment by secana** [comment]
Fixed in Kellnr 4.0.3
